src/components/NetworkSelection/NetworkSelection.tsx

```diff
@@ -0,0 +1,90 @@
+import { ALL_CHAINS, Chain } from '@pangolindex/sdk';
+import React, { useState } from 'react';
+import { Scrollbars } from 'react-custom-scrollbars';
+import { Box } from 'src/components/Box';
+import Modal from 'src/components/Modal';
+import { Text } from 'src/components/Text';
+import { ToggleButtons } from 'src/components/ToggleButtons';
+import { ButtonFrame, ChainButton, ChainsList, CloseButton, Frame, Logo } from './styled';
+import { NetworkProps } from './types';
+
+interface MetamaskError {
+  code: number;
+  message: string;
+}
+
+const NetworkSelection: React.FC<NetworkProps> = (props) => {
+  const { open, closeModal } = props;
+  const [mainnet, setMainnet] = useState(true);
+
+  const { ethereum } = window;
+  const isMetaMask = ethereum && ethereum.isMetaMask;
+
+  const chains = ALL_CHAINS.filter((chain) => chain.pangolin_is_live && chain.mainnet === mainnet);
+
+  const chainListHeight = chains.length / 2 <= 1 ? 48 : chains.length / 2 <= 2 ? 116 : 184;
+
+  const changeChain = async (chain: Chain) => {
+    if (isMetaMask && ethereum) {
+      try {
+        await ethereum.request({
+          method: 'wallet_switchEthereumChain',
+          params: [{ chainId: `0x${chain.chain_id.toString(16)}` }],
+        });
+      } catch (error) {
+        // This error code indicates that the chain has not been added to MetaMask.
+        const metamask = error as MetamaskError;
+        if (metamask.code === 4902) {
+          await ethereum.request({
+            method: 'wallet_addEthereumChain',
+            params: [
+              {
+                chainName: chain.name,
+                chainId: `0x${chain.chain_id.toString(16)}`,
+                //nativeCurrency: chain.nativeCurrency,
+                rpcUrls: [chain.rpc_uri],
+                blockExplorerUrls: chain.blockExplorerUrls,
+                iconUrls: chain.logo,
+                nativeCurrency: chain.nativeCurrency,
+              },
+            ],
+          });
+        }
+      }
+    }
+  };
+
+  return (
+    <Modal isOpen={open} onDismiss={closeModal}>
+      <Frame>
+        <CloseButton onClick={closeModal} />
+        <Text color="text1" fontSize="24px" marginBottom={20} style={{ gridArea: 'text' }}>
+          Select Chain
+        </Text>
+        <ButtonFrame>
+          <ToggleButtons
+            options={['Mainnet', 'Testnet']}
```

Comment:
lets move this to enum something like
`options={[NETWORK_TYPE.MAINNET, NETWORK_TYPE.TESTNET]}`

and use that enum everywhere instead of direct string